app/assets/javascripts/discourse/tests/integration/components/admin-schema-theme-setting/editor-test.gjs

```diff
@@ -355,5 +283,138 @@ module(
         })
       );
     });
+
+    test("input fields for items at different levels", async function (assert) {
+      const [schema, data] = schemaAndData(2);
+      await render(<template>
+        <AdminSchemaThemeSettingEditor @schema={{schema}} @data={{data}} />
+      </template>);
+
+      const container = new InputFieldsFromDOM();
+
+      assert.strictEqual(container.count, 2);
+      assert.dom(container.fields.name.labelElement).hasText("name");
+      assert.dom(container.fields.icon.labelElement).hasText("icon");
+
+      assert.dom(container.fields.name.inputElement).hasValue("nice section");
+      assert.dom(container.fields.icon.inputElement).hasValue("arrow");
+
+      const tree = new TreeFromDOM();
+      await click(tree.nodes[1].element);
+
+      container.refresh();
+      tree.refresh();
+
+      assert.strictEqual(container.count, 2);
+      assert.dom(container.fields.name.labelElement).hasText("name");
+      assert.dom(container.fields.icon.labelElement).hasText("icon");
+
+      assert.dom(container.fields.name.inputElement).hasValue("cool section");
+      assert.dom(container.fields.icon.inputElement).hasValue("bell");
+
+      await click(tree.nodes[1].children[0].element);
+
+      tree.refresh();
+      container.refresh();
+
+      assert.strictEqual(container.count, 3);
+      assert.dom(container.fields.text.labelElement).hasText("text");
+      assert.dom(container.fields.url.labelElement).hasText("url");
+      assert.dom(container.fields.icon.labelElement).hasText("icon");
+
+      assert.dom(container.fields.text.inputElement).hasValue("About");
+      assert
+        .dom(container.fields.url.inputElement)
+        .hasValue("https://example.com/about");
+      assert.dom(container.fields.icon.inputElement).hasValue("asterisk");
+    });
+
+    test("identifier field instantly updates in the navigation tree when the input field is changed", async function (assert) {
+      const [schema, data] = schemaAndData(2);
+      await render(<template>
+        <AdminSchemaThemeSettingEditor @schema={{schema}} @data={{data}} />
+      </template>);
+
+      const container = new InputFieldsFromDOM();
+      const tree = new TreeFromDOM();
+
+      await fillIn(
+        container.fields.name.inputElement,
+        "nice section is really nice"
+      );
+
+      assert.dom(tree.nodes[0].element).hasText("nice section is really nice");
+
+      await click(tree.nodes[0].children[0].element);
+
+      container.refresh();
+      tree.refresh();
+
+      await fillIn(
+        container.fields.text.inputElement,
+        "Security instead of Privacy"
+      );
+
+      assert.dom(tree.nodes[0].element).hasText("Security instead of Privacy");
+    });
+
+    test("edits are remembered when navigating between levels", async function (assert) {
```

Comment:
Just a small note here but with jsonb we can actually do partial updates instead of having to rewrite the whole row in the database just to update a single object. I wonder if it'll simply things if we treat each level as its own object and prevent updating multiple objects in one goal?

Reply:
Hmmm yeah I think that will simplify things on the backend, but we've not fully figured out how the UX will be like so this will probably change when we get to test the interface when it's fleshed out more.